test/libsolidity/astPropertyTests/forLoop/isSimpleCounterLoop_negative_cases.sol

```diff
@@ -0,0 +1,134 @@
+type UINT is uint;
+
+function add(UINT x, UINT y) pure returns (UINT) {
+    return UINT.wrap(UINT.unwrap(x) + UINT.unwrap(y));
+}
+
+function lt(UINT x, UINT y) pure returns (bool) {
+    return UINT.unwrap(x) < UINT.unwrap(y);
+}
+
+using {lt as <, add as +} for UINT global;
+
+function g() pure returns (bool) {
+    return false;
+}
+
+function h() pure returns (uint) {
+    return 13;
+}
+
+contract C {
+    uint z = 0;
+
+    function modifyStateVarZ() public returns (uint) {
+        z = type(uint).max;
+        return z;
+    }
+
+    function f() public {
```

Comment:
We should probably add some tests using the shorthand notation for binary operations, e.g. `i += 1` in the post section of the loop statement.

Reply:
Added.